dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/export/openaire/OpenAireExportUtil.java

```diff
@@ -1577,4 +1540,68 @@ private static boolean isValidEmailAddress(String email) {
         }
         return result;
     }
+
+    private static void writeRightsUriLicenseInfoAttribute(XMLStreamWriter xmlw, List<FileDTO> files, String language) throws XMLStreamException {
+        if(!Lists.isEmpty(files)) {
+            writeRightsHeader(xmlw, language);
+
+            if(areAllFilesRestricted(files)) {
+                xmlw.writeCharacters("Access to all files in the dataset is restricted.");
+            } else if(areAllFilesAllRightsReserved(files)) {
+                xmlw.writeCharacters("All rights reserved.");
+            } else if(areAllFilesUnderSameLincese(files)) {
+                xmlw.writeAttribute("rightsURI", files.get(0).getLicenseUrl());
+                xmlw.writeCharacters(files.get(0).getLicenseName());
+            } else if(hasRestrictedFile(files)) {
+                xmlw.writeCharacters("Different licenses and/or terms apply to individual files in the dataset. Access to some files in the dataset is restricted.");
+            } else {
+                xmlw.writeCharacters("Different licenses and/or terms apply to individual files in the dataset.");
+            }
+
+            xmlw.writeEndElement();
+        }
+    }
+
+    private static boolean areAllFilesUnderSameLincese(List<FileDTO> files) {
+        final String firstFileLicense = files.get(0).getLicenseName();
+        if(StringUtils.isNotEmpty(firstFileLicense)) {
+            return files
+                    .stream()
+                    .allMatch(fileDTO -> fileDTO.getLicenseName().equals(firstFileLicense));
```

Comment:
you can also do 
```
allMatch(files.get(0).getLicenseName()::equals);
```
Both are readable for me so just a suggestion.

Reply:
`files.get(0).getLicenseName()` is a `String`, while stream operates on `FileDTO` objects. It should work that way.

Reply:
Yeah you also would need to map license name before in stream.

Reply:
Not going to make code more complicated than necessary :P